lib/linked_list.rb

```diff
@@ -10,36 +10,71 @@ def initialize
   # Time complexity - ?
   # Space complexity - ?
   def add_first(data)
-
+    @head = Node.new(data, @head)
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def get_first
-
+    if @head.nil?
+      return nil
+    else
+      return @head.data
+    end
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def length
-    return 0
+    count = 0
+    temp = @head
+    while temp != nil
+      count += 1
+      temp = temp.next
+    end
+    return count
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def add_last(data)
```

Comment:
👍 
This works, but it's a little overly complex.  Can you see a way to simplify it?